Documentation/proposals/202310-automated-sharding.md

```diff
@@ -0,0 +1,130 @@
+# Automated Sharding
```

Comment:
I'm thinking about!
Maybe we should switch to Sharding AutoScaling.
Looking to Automated sharding, seems we're going to automate the sharding decision for users.
Like automatically see their instances are reaching a specific size and shard it out-of-box.

wdyt?